block/sync.go

```diff
@@ -0,0 +1,121 @@
+package block
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/dymensionxyz/dymint/node/events"
+	"github.com/dymensionxyz/dymint/settlement"
+	uevent "github.com/dymensionxyz/dymint/utils/event"
+	"github.com/dymensionxyz/gerr-cosmos/gerrc"
+	"github.com/tendermint/tendermint/libs/pubsub"
+)
+
+// onNewStateUpdate will update the last submitted height and will update sequencers list from SL. After, it triggers syncing or validation, depending whether it needs to sync first or only validate.
+func (m *Manager) onNewStateUpdate(event pubsub.Message) {
+	eventData, ok := event.Data().(*settlement.EventDataNewBatch)
+	if !ok {
+		m.logger.Error("onReceivedBatch", "err", "wrong event data received")
+		return
+	}
+
+	// Update heights based on state update end height
+	m.LastSubmittedHeight.Store(eventData.EndHeight)
```

Comment:
can we rename LastSubmittedHeight to last hub height or similar

Reply:
done